### PR TITLE
[ISPN-7802] Limit buffer sizes when reading/writing

### PR DESCRIPTION
**Pull request by dereed**
https://issues.jboss.org/browse/ISPN-7802

### CODE REVIEW

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/TcpTransport.java [hunk 1]
```diff
@@ -158,24 +158,18 @@ public int readVInt() {
...
-      }
+      writeBytes(toAppend, 0, toAppend.length);
    }
 
    @Override
    protected void writeBytes(byte[] toAppend, int offset, int count) {
       try {
-         socketOutputStream.write(toAppend, offset, count);
+         for ( int o = offset; o < offset + count; o += SOCKET_STREAM_BUFFER )
+         {
```
Comment (danberindei) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/TcpTransport.java]: Brace should be on the same line as the `if`.

**Comment by danberindei** [comment]
@dereed Do you mind if I integrate this instead?

https://github.com/infinispan/infinispan/commit/2ab2269e18f2f918f5cff8626f5efc6a2988233f

**Comment by dereed** [comment]
Yes, go ahead.

-Dennis

**Comment by wburns** [comment]
@danberindei it looks like we need a new PR with your commit then?

**Comment by danberindei** [comment]
@wburns I just pushed my commit, I had forgotten about it.

**Comment by danberindei** [comment]
Pushed to 9.0.x as well.

**Pull request closed without merging**